pkg/datarepair/checker/checker.go

```diff
@@ -75,34 +77,17 @@ func NewChecker(metainfo *metainfo.Service, repairQueue queue.RepairQueue, overl
 func (checker *Checker) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	c := make(chan error)
+	group, ctx := errgroup.WithContext(ctx)
 
-	go func() {
-		c <- checker.Loop.Run(ctx, func(ctx context.Context) error {
-			err := checker.IdentifyInjuredSegments(ctx)
-			if err != nil {
-				checker.logger.Error("error with injured segments identification: ", zap.Error(err))
-			}
-			return nil
-		})
-	}()
+	group.Go(func() error {
+		return errs2.IgnoreCanceled(checker.InjuredSegmentCron(ctx))
+	})
 
-	go func() {
-		c <- checker.IrreparableLoop.Run(ctx, func(ctx context.Context) error {
-			err := checker.IrreparableProcess(ctx)
-			if err != nil {
-				checker.logger.Error("error with irreparable segments identification", zap.Error(err))
-			}
-			return nil
-		})
-	}()
+	group.Go(func() error {
+		return errs2.IgnoreCanceled(checker.IrreparableSegmentCron(ctx))
```

Comment:
you could probably use an inline / anonymous function here (similar to how you had it before) and increase readability.

Reply:
updated.